src/app/phase2/issue/issue.component.ts

```diff
@@ -1,84 +1,44 @@
-import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { Issue } from '../../core/models/issue.model';
 import { IssueService } from '../../core/services/issue.service';
-import { FormBuilder } from '@angular/forms';
-import { finalize } from 'rxjs/operators';
-import { ErrorHandlingService } from '../../core/services/error-handling.service';
-import { IssueCommentService } from '../../core/services/issue-comment.service';
-import { IssueComment, IssueComments } from '../../core/models/comment.model';
 import { PermissionService } from '../../core/services/permission.service';
-import { Subscription } from 'rxjs';
+import { ViewIssueComponent, ISSUE_COMPONENTS } from '../../shared/view-issue/view-issue.component';
 
 @Component({
   selector: 'app-issue',
   templateUrl: './issue.component.html',
   styleUrls: ['./issue.component.css']
 })
-export class IssueComponent implements OnInit, OnDestroy {
-  issue: Issue;
-  comments: IssueComment[];
-  isIssueLoading = true;
-  isCommentsLoading = true;
-  isResponseEditing = false;
-  isDescriptionEditing = false;
-  issueSubscription: Subscription;
+export class IssueComponent implements OnInit {
+  issueId: number;
+
+  readonly issueComponents: ISSUE_COMPONENTS[] = [
+    ISSUE_COMPONENTS.TESTER_POST,
+    ISSUE_COMPONENTS.TEAM_RESPONSE,
+    ISSUE_COMPONENTS.NEW_TEAM_RESPONSE,
+    ISSUE_COMPONENTS.SEVERITY_LABEL,
+    ISSUE_COMPONENTS.TYPE_LABEL,
+    ISSUE_COMPONENTS.RESPONSE_LABEL,
+    ISSUE_COMPONENTS.ASSIGNEE,
+    ISSUE_COMPONENTS.DUPLICATE
+  ];
+
+  @ViewChild(ViewIssueComponent) viewIssue: ViewIssueComponent;
 
   constructor(public issueService: IssueService,
-              private issueCommentService: IssueCommentService,
               private route: ActivatedRoute,
-              private formBuilder: FormBuilder,
-              private errorHandlingService: ErrorHandlingService,
               public permissions: PermissionService) { }
 
   ngOnInit() {
     this.route.params.subscribe(
       params => {
-        const id = +params['issue_id'];
-        this.initializeIssue(id);
-        this.initializeComments(id);
+        this.issueId = + params['issue_id'];
```

Comment:
@JunWei96 A query, what's the purpose of the `= +` ? I'm unfamiliar with this syntax in TypeScript and couldn't find any documentation while searching for the `TypeScript Assignment Operators`.

Reply:
And removing it caused the app to behave unexpectedly 😅

Reply:
Here is an article about it!
https://medium.com/@nikjohn/cast-to-number-in-javascript-using-the-unary-operator-f4ca67c792ce

Reply:
Ahh thanks!